### ISSUE TITLE
ServiceControl does not deduplicate audit messages

### ISSUE THREAD

**Issue by SzymonPobiega** [Particular/ServiceControl#1056]
### Symptoms

In case of network issues audit messages might get duplicated and, as a result, ServiceInsight shows multiple copies of the same message,

### Who's affected

Endpoints in `SendsAtomicWithReceive` and lower transaction modes. All Outbox-enabled endpoints are affected.

**Comment by SzymonPobiega** [comment]
@Particular/servicecontrol-maintainers what do you think?

**Comment by WilliamBZA** [comment]
I think this can be left as is for now. The performance impact is not worth a release that changes existing behaviour. This must however be addressed in the technical debt work.

**Comment by SzymonPobiega** [comment]
![duplicate message si](https://user-images.githubusercontent.com/163318/34200136-48824786-e570-11e7-9ba0-fd4fa63abed3.png)

@mikeminutillo @WilliamBZA here's the SI view. I generated it by sending a duplicate message to the outbox queue using this behavior:

```
class DuplicateBehavior : Behavior<IDispatchContext>
{
    public override async Task Invoke(IDispatchContext context, Func<Task> next)
    {
        await next();
        await next();
    }
}
```

**Comment by SzymonPobiega** [comment]
Maybe we can do the deduplication by message ID when we do scatter-gather queries?

**Comment by mikeminutillo** [comment]
Given that we have updated Raven and NSB, should we consider putting a deterministic Id in place again? A hash of Message Id, Endpoint, and Time Processed would do.

**Comment by SzymonPobiega** [comment]
@mikeminutillo sounds like a simple enough solution to qualify for MR

**Comment by SzymonPobiega** [comment]
Fixed in #2154
